pkg/registration/spoke/registration/creating_controller.go

```diff
@@ -112,15 +116,33 @@ func skipUnauthorizedError(err error) error {
 	return err
 }
 
+// AnnotationDecorator set annotations from annotation map to ManagedCluster
 func AnnotationDecorator(annotations map[string]string) ManagedClusterDecorator {
 	return func(cluster *clusterv1.ManagedCluster) *clusterv1.ManagedCluster {
-		filteredAnnotations := commonhelpers.FilterClusterAnnotations(annotations)
 		if cluster.Annotations == nil {
 			cluster.Annotations = make(map[string]string)
 		}
-		for key, value := range filteredAnnotations {
-			cluster.Annotations[key] = value
+
+		filteredAnnotations := commonhelpers.FilterClusterAnnotations(annotations)
+		for k, v := range filteredAnnotations {
+			cluster.Annotations[k] = v
 		}
+
+		return cluster
+	}
+}
+
+// LabelDecorator set labels from label map to ManagedCluster
+func LabelDecorator(labels map[string]string) ManagedClusterDecorator {
+	return func(cluster *clusterv1.ManagedCluster) *clusterv1.ManagedCluster {
+		if cluster.Labels == nil {
+			cluster.Labels = make(map[string]string)
+		}
+
+		for k, v := range labels {
```

Comment:
note we should not allowing set ANY label from the spoke. For instance, open-cluster-management.io/clusterset label is to define the label on the managedcluster, it should not be set by agent.